From eedc8f34edbe6d02f6587532b40c352bded08259 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Sun, 14 Dec 2014 11:03:09 +0100
Subject: [PATCH 4/4] Monolithic: Don't go online on startup when there is no
 usable adapter

Fix issue with Bluetooth icon visible in systray when Bluetooth is
offline (no usable adapter).

BUG: 341768
---
 src/monolithic/monolithic.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/monolithic/monolithic.cpp b/src/monolithic/monolithic.cpp
index 86bb9d4..3ce46ac 100644
--- a/src/monolithic/monolithic.cpp
+++ b/src/monolithic/monolithic.cpp
@@ -44,7 +44,7 @@ Monolithic::Monolithic(QObject* parent)
 
     offlineMode();
 
-    if (!Manager::self()->adapters().isEmpty()) {
+    if (Manager::self()->usableAdapter()) {
         onlineMode();
     }
 
@@ -55,7 +55,6 @@ Monolithic::Monolithic(QObject* parent)
     setStandardActionsEnabled(false);
     setAssociatedWidget(contextMenu());
 
-    setStatus(KStatusNotifierItem::Active);
     poweredChanged();
 }
 
-- 
2.1.0

